todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.kt

```diff
@@ -80,22 +81,20 @@ class TasksLocalDataSource private constructor(context: Context) : TasksDataSour
         val cursor = db.query(
                 TABLE_NAME, projection, "$COLUMN_NAME_ENTRY_ID LIKE ?", arrayOf(taskId), null,
                 null, null)
-
+        var task: Task? = null
         with(cursor) {
             if (moveToFirst()) {
                 val itemId = getString(getColumnIndexOrThrow(COLUMN_NAME_ENTRY_ID))
                 val title = getString(getColumnIndexOrThrow(COLUMN_NAME_TITLE))
                 val description = getString(getColumnIndexOrThrow(COLUMN_NAME_DESCRIPTION))
-                val task = Task(title, description, itemId).apply {
+                 task = Task(title, description, itemId).apply {
                     isCompleted = getInt(getColumnIndexOrThrow(COLUMN_NAME_COMPLETED)) == 1
                 }
-                callback.onTaskLoaded(task)
-            } else {
-                callback.onDataNotAvailable()
             }
             close()
         }
         db.close()
+        task?.let { callback.onTaskLoaded(it) } ?: callback.onDataNotAvailable()
```

Comment:
Hmm, although this is working as intended, I'm not sure this is best practice.
For example:

val test = `expression A`?.let { `expression B` } ?: `expression C`

If `expression A` evaluates to a nonnull and `expression B` evaluates to null, then the value of the left side of the elvis operator is null (because the return value for a `let` block is the last expression inside the block), so `test` would be set to `expression C`. The intention was to set `test` to `expression C` only if `expression A` is null, so that would be incorrect.

Keeping the original would probably be the safest way to go.